src/chains/filecoin/options/tests/logging-options.test.ts

```diff
@@ -0,0 +1,177 @@
+          const file = resolve("./eperm-file.log");
+          try {
+            const handle = await open(file, "w");
+            // set no permissions on the file
+            await handle.chmod(0);
+            await handle.close();
+
+            const error = {
+              message: `Failed to open log file ${file}. Please check if the file path is valid and if the process has write permissions to the directory.`
+            };
+
+            assert.throws(
+              () =>
+                FilecoinOptionsConfig.normalize({
+                  logging: { file }
+                }),
+              error
+            );
+          } finally {
+            await unlink(file);
+          }
+        });
+
+        it("should append to the specified file", async () => {
+          const message = "message";
+          const handle = await open(validFilePath, "w");
+          try {
+            await handle.write("existing content");
+            handle.close();
+
+            const options = FilecoinOptionsConfig.normalize({
+              logging: { file: validFilePath }
+            });
+            options.logging.logger.log(message);
+            closeSync(options.logging.file);
+
+            const readHandle = await open(validFilePath, "r");
+            const content = await readHandle.readFile({ encoding: "utf8" });
+            await readHandle.close();
+            assert(
+              content.startsWith("existing content"),
+              "Existing content was overwritten by the logger"
+            );
+          } finally {
+            await unlink(validFilePath);
+          }
+        });
+
+       it("uses the provided logger, and file when both `logger` and `file` are provided", async () => {
+          const calls: any[] = [];
+          const logger = {
+            log: (message: any, ...params: any[]) => {
+              calls.push([message, ...params]);
+            }
+          };
+
+          try {
+            const options = FilecoinOptionsConfig.normalize({
+              logging: {
+                logger,
+                file: validFilePath
+              }
+            });
+
+            options.logging.logger.log("message", "param1", "param2");
+            if ("getCompletionHandle" in options.logging.logger) {
+              //wait for the logs to be written to disk
+              await options.logging.logger.getCompletionHandle();
+            } else {
+              throw new Error("Expected options.logging.logger to be AsyncronousLogger");
+            }
+            closeSync(options.logging.file);
+
+            assert.deepStrictEqual(calls, [["message", "param1", "param2"]]);
+
+            const fromFile = await readFile(validFilePath, "utf8");
+            assert(fromFile !== "", "Nothing written to the log file");
+
+            const timestampPart = fromFile.substring(0, 24);
+
+            const timestampRegex =
+              /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z$/;
```

Comment:
```suggestion
              /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}Z$/;
```

Restrict wildcard match